### PR TITLE
Improve general code quality

### PR DESCRIPTION
**Pull request by hug-dev**
This commit:
* uses a ci.sh script to perform all CI tasks
* adds cross-compilation logic
* improves Slot to integer conversion handling (thanks to
@george-hopkins)
* adds some deny
* implements Copy, Debug, Clone where needed
* moves tests to the integration test folder
* fixes various lints
* adds credit

Co-authored-by: George Hopkins <george-hopkins@null.net>
Signed-off-by: Hugues de Valon <hugues.devalon@arm.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,6 +14,14 @@ Please check the [**Contribution
 Guidelines**](https://parallaxsecond.github.io/parsec-book/contributing/index.html) to know more
 about the contribution process.
 
+# History
+
+This repository is based on [this original PR on rust-pkcs11](https://github.com/mheese/rust-pkcs11/pull/43).
+Read the PR discussion for more information.
+
+Thanks to Joe Ellis (@joechrisellis) and Ionuț Mihalcea (@ionut-arm) for their early work on the
+crate.
```
Comment (ionut-arm) [inline README.md:23]: At least add yourself as well - "Joe Ellis, Ionut Mihalcea, and Hugues de Valon developed the early prototype...". Seems weird to only mention us two when it's not on your personal repo.

File: cryptoki-sys/build.rs [hunk 2]
```diff
@@ -6,7 +6,12 @@ fn main() {
 
     #[cfg(not(feature = "generate-bindings"))]
     {
-        let supported_platforms = vec![String::from("x86_64-unknown-linux-gnu")];
+        let supported_platforms = vec![
+            "x86_64-unknown-linux-gnu".to_string(),
+            "aarch64-unknown-linux-gnu".to_string(),
+            "armv7-unknown-linux-gnueabi".to_string(),
+            "arm-unknown-linux-gnueabi".to_string(),
+        ];
```
Comment (ionut-arm) [inline cryptoki-sys/build.rs:14]: 🚀

File: cryptoki/tests/common.rs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+use cryptoki::types::session::UserType;
+use cryptoki::types::slot_token::Slot;
+use cryptoki::types::Flags;
+use cryptoki::Pkcs11;
+use std::env;
+
+pub fn init_pins() -> (Pkcs11, Slot) {
+    let pkcs11 = Pkcs11::new(
+        env::var("PKCS11_SOFTHSM2_MODULE")
+            .unwrap_or_else(|_| "/usr/local/lib/softhsm/libsofthsm2.so".to_string()),
```
Comment (ionut-arm) [inline cryptoki/tests/common.rs:11]: Nice! 💯

File: cryptoki/tests/basic.rs [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+use cryptoki::types::mechanism::Mechanism;
+use cryptoki::types::object::{Attribute, AttributeInfo, AttributeType, KeyType, ObjectClass};
+use cryptoki::types::session::UserType;
+use cryptoki::types::Flags;
+use serial_test::serial;
+use std::sync::Arc;
+use std::thread;
+
+#[test]
+#[serial]
```
Comment (ionut-arm) [inline cryptoki/tests/basic.rs:13]: Nice, didn't know about this before

**Review by ionut-arm — APPROVED** [review]
This is great! Cheers

File: README.md (hunk 1 above)
Comment (ionut-arm) [reply README.md:23]: Alternatively just don't mention anyone - `Parsec developers created ....` though that's kinda superfluous

Comment (hug-dev) [reply README.md:23]: hmm yeah I wanted to credit those involved in this... I will just add them as co-author of this specific commit.

Comment (hug-dev) [reply README.md:23]: (I should have done that for the first commit but I forgot)

**Pull request merged**